voluncheer/opportunityboard/views/search.py

```diff
@@ -9,12 +9,73 @@
 from opportunityboard.views.opportunityboard import category_dict_gen
 
 
+class Filter:
+    def __init__(
+        self,
+        category=None,
+        subcategory=None,
+        subsubcategory=None,
+        duration=None,
+        distance=0,
+        start_date=None,
+    ) -> None:
+        self.category = category
+        self.subcategory = subcategory
+        self.subsubcategory = subsubcategory
+        self.duration = duration
+        self.distance = distance
+        self.start_date = start_date
+
+    def gen_dict(self):
+        return {
+            "category": self.category,
+            "subcategory": self.subcategory,
+            "subsubcategory": self.subsubcategory,
+            "duration": self.duration,
+            "distance": self.distance,
+            "start_date": self.start_date,
+        }
+
+    def gen_category_placeholder(self):
+        if self.category is not None:
+            return self.category
+        elif self.subcategory is not None:
+            return self.subcategory
+        elif self.subsubcategory is not None:
+            return self.subsubcategory
+        return "CATEGORY"
+
+    def gen_duration_placeholder(self):
+        if self.duration is not None:
+            return self.duration
+        return "DURATION"
+
+    def search(self):
+        """Search by the given filters
+        Input: Filter object
+        category(no-filter: None)
+        subcategory(no-filter: None)
+        subsubcategory(no-filter: None)
+        Output: Opportunity list
+        """
+        filtered_opportunity = Opportunity.objects.all()
+        if self.category is not None:
+            filtered_opportunity = filtered_opportunity.filter(category=self.category)
+        if self.subcategory is not None:
+            filtered_opportunity = filtered_opportunity.filter(subcategory=self.subcategory)
+        if self.subsubcategory is not None:
+            filtered_opportunity = filtered_opportunity.filter(subsubcategory=self.subsubcategory)
+
+        return filtered_opportunity
+
+
 def filter_search(request):
+    filter = Filter()
     if request.method == "POST":
         filter = parse_search_filter(request.POST)
         logger = logging.getLogger(__name__)
-        logger.error(filter)
-        opportunity_lists = search_by_filters(filter)
+        logger.error(filter.gen_dict())
+        opportunity_lists = filter.search()
     else:
```

Comment:
This is essentially a repeat of the opportunityboard view right? Could we use `return render HttpRedirectResponse(reverse(opportunityboard))` instead? Or is there some way we can avoid repetition?

Also, since we are not modifying/inserting data, can we use GET instead of POST? In that case, we change the form to GET and then there is only one branch for `search()`? Just an idea.